engine/src/main/java/org/hawkular/alerts/engine/impl/ispn/IspnDefinitionsServiceImpl.java

```diff
@@ -489,7 +917,48 @@ public void removeDampening(String tenantId, String dampeningId) throws Exceptio
 
     @Override
     public void removeGroupDampening(String tenantId, String groupDampeningId) throws Exception {
+        if (isEmpty(tenantId)) {
+            throw new IllegalArgumentException("TenantId must be not null");
+        }
+        if (isEmpty(groupDampeningId)) {
+            throw new IllegalArgumentException("dampeningId must be not null");
+        }
 
+        try {
+            deferNotifications();
+
+            Dampening groupDampening = null;
+            try {
+                groupDampening = getDampening(tenantId, groupDampeningId);
+            } catch (NotFoundException e) {
+                log.debugf("Ignoring removeDampening(%s), the Dampening does not exist.", groupDampeningId);
+                return;
+            }
+
+            String groupId = groupDampening.getTriggerId();
+            // fetch the trigger (or throw NotFoundException)
+            Trigger groupTrigger = getTrigger(tenantId, groupId);
+            if (!groupTrigger.isGroup()) {
+                throw new IllegalArgumentException(
+                        "Trigger [" + tenantId + "/" + groupId + "] is not a group trigger.");
+            }
+
+            Collection<Trigger> memberTriggers = getMemberTriggers(tenantId, groupId, false);
+
+            for (Trigger member : memberTriggers) {
+                Collection<Dampening> dampenings = getTriggerDampenings(tenantId, member.getId(),
+                        groupDampening.getTriggerMode());
+                if (dampenings.isEmpty()) {
+                    continue;
+                }
+                removeDampening(dampenings.iterator().next());
```

Comment:
It can return multiple if the 'triggerMode' param is null: From jdoc:
Parameters:
  triggerMode Return only dampenings for the given trigger mode. Return all if null.